src/main/java/com/puppycrawl/tools/checkstyle/ant/CheckstyleAntTask.java

```diff
@@ -529,19 +541,16 @@ private List<File> scanPath(Path path, int pathIndex) {
         log(pathIndex + ") Scanning path " + path, Project.MSG_VERBOSE);
         final List<File> allFiles = new ArrayList<>();
         int concreteFilesCount = 0;
-
         for (String resource : resources) {
             final File file = new File(resource);
             if (file.isFile()) {
                 concreteFilesCount++;
                 allFiles.add(file);
             }
             else {
-                final DirectoryScanner scanner = new DirectoryScanner();
```

Comment:
Reusing this might also improve performance. At the very least, it reduces duplication and centralizes what can be reused.